### PR TITLE
Unit tests - Tests for `wdb_close`, `wdb_leave` and `wdb_finalize_all_statements`

### PR DESCRIPTION
**Pull request by san1526**
|Related issue|
|---|
|#11754|

## Description
Adding unit tests for new wdb functions` wdb_close`, `wdb_leave` and `wdb_finalize_all_statements`

- Compilation without warnings in every supported platform
  - [x] Linux

- Memory tests for Linux
  - [x] Scan-build report
  - [x] AddressSanitizer

- [x] Added unit tests (for new features)

### CODE REVIEW

**Comment by san1526** [comment]
image
image

File: src/unit_tests/wazuh_db/test_wdb.c [hunk 1]
```diff
@@ -752,6 +763,143 @@ void test_wdb_get_config(){
...
+    assert_string_equal("[\"COL_TEXT_0\",\"COL_TEXT_1\",\"COL_TEXT_2\",\"COL_TEXT_3\"]", ret_str);
+
+    cJSON_Delete(ret);
+    free(ret_str);
+}
+
+void test_wdb_exec_single_column_invalid_stmt(){
+    expect_string(__wrap__mdebug1, formatted_msg, "Invalid SQL statement.");
+
+    cJSON *ret = wdb_exec_stmt_single_column(0);
```
Comment (palaciosjeremias) [inline src/unit_tests/wazuh_db/test_wdb.c:794]: Just for clarity, we can replac 0 with NULL as the argument is a pointer.

File: src/unit_tests/wazuh_db/test_wdb.c [hunk 2]
```diff
@@ -752,6 +763,143 @@ void test_wdb_get_config(){
     cJSON_Delete(ret);
 }
```
Comment (palaciosjeremias) [inline src/unit_tests/wazuh_db/test_wdb.c:765]: This is legacy, but we should add a comment like "/* wdb_exec_stmt_single_column */" as a header for the block of tests for the same method.

File: src/unit_tests/wazuh_db/test_wdb.c [hunk 3]
```diff
@@ -752,6 +763,143 @@ void test_wdb_get_config(){
...
+void test_wdb_finalize_all_statements(){
+    const int kMaxStmt = 10;
+    wdb_t wdb = {0};
+
+    for (int i = 0; i < kMaxStmt; ++i) { wdb.stmt[i] = (sqlite3_stmt *)0xDEADBEEF; }
+
+    struct stmt_cache_list** c = &(wdb.cache_list);
+    for(int i = 0; i < kMaxStmt; ++i){
+        *c = calloc(1, sizeof(struct stmt_cache_list));
+
```
Comment (palaciosjeremias) [inline src/unit_tests/wazuh_db/test_wdb.c:832]: Code style: I think we can eliminate extra white spaces here.

File: src/unit_tests/wazuh_db/test_wdb.c [hunk 4]
```diff
@@ -752,6 +763,143 @@ void test_wdb_get_config(){
...
+        (*c)->value.stmt = (sqlite3_stmt*) 0xDEADBEEF;
+
+        c = &((*c)->next);
+    }
+
+    *c = 0;
+
+    // free the prepared statements
+    will_return_count(__wrap_sqlite3_finalize, 1, kMaxStmt);
+
```
Comment (palaciosjeremias) [inline src/unit_tests/wazuh_db/test_wdb.c:842]: Code style: I think we can eliminate extra white spaces here.

File: src/unit_tests/wazuh_db/test_wdb.c [hunk 5]
```diff
@@ -752,6 +763,143 @@ void test_wdb_get_config(){
...
+    will_return(__wrap_sqlite3_close_v2, SQLITE_ERROR);
+    will_return(__wrap_sqlite3_errmsg, "mock_error");
+
+    expect_string(__wrap__merror, formatted_msg, "DB(agent) wdb_close(): mock_error");
+
+    assert_int_equal(-1, wdb_close(&wdb, 0));
+}
+
+void test_wdb_close(){
```
Comment (palaciosjeremias) [inline src/unit_tests/wazuh_db/test_wdb.c:879]: We use to call this test as test_wdb_close_success()

File: src/wazuh_db/wdb.c [hunk 6]
```diff
@@ -1278,6 +1278,8 @@ void wdb_finalize_all_statements(wdb_t * wdb) {
         os_free(node_stmt);
         node_stmt = temp;
     }
+
+    wdb->cache_list = NULL;
```
Comment (palaciosjeremias) [inline src/wazuh_db/wdb.c:1282]: Good catch!

**Review by palaciosjeremias — APPROVED** [review]
Great Job!
Just minor code style suggestions.

**Pull request merged**